### PR TITLE
Add the direct upload to S3 chapter and reference to documentation

### PR DESCRIPTION
**Pull request by cilim**
Kako smo pricali na backend weeklyu, evo chapter s referencom na eksterni sajt koji ima super dokumentaciju o direct uploadu na S3, koristeci Shrine gem.

Task: https://app.productive.io/1-infinum/tasks/m/task/123918?filter=LTE%3D

### CODE REVIEW

File: S3 Direct Upload.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# General
+
+S3 direct upload is the process of uploading files to the Amazon's S3 storage service.
+It is highly useful if you want to speed up the uploading process and to improve user experience.
+
+# The Workflow
+
+[This documentation](https://www.ironin.it/blog/store-your-files-on-s3-using-the-ruby-shrine-gem-part-2.html)
+covers all the code and configuration files you need for efficient direct upload to S3.
```
Comment (vr4b4c) [inline S3 Direct Upload.md]: promijenio bi
`...you need for efficient direct upload to S3. ...`
u
`needed for supporting direct upload to S3`

Comment (cilim) [reply S3 Direct Upload.md]: Da, bolje zvuci!

Thanks! 👍

**Comment by DamirSvrtan** [comment]
I'd say this guideline misses the WHY factor. What's the motivation behind direct upload? Why would you use it? I'd just state something like this ~

> Often times mobile or web frontends upload files through the app server which means that the file does a double hop: from the frontend to the backend, then from the backend to S3.
> 
> Direct upload resolves this double hop performance problem by handing out one-time credentials to the frontend app to upload files directly to S3 and sends out references to those files to the backend.
>

**Comment by cilim** [comment]
@DamirSvrtan I updated the chapter with your input. Well written! :)

**Comment by d4be4st** [comment]
This can actually be used on any cloud storage, it does not have to be S3, although we use S3.



So i would make it a bit more generic, and add at the bottom that we use S3 for our purposes

**Comment by cilim** [comment]
@d4be4st chapter updated with your suggestion. Thanks!

**Comment by d4be4st** [comment]
Would add jankos post about this topic https://twin.github.io/better-file-uploads-with-shrine-direct-uploads/

**Pull request merged**